src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/ConsiderPassingBaseTypesAsParametersTests.cs

```diff
@@ -0,0 +1,418 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.ConsiderPassingBaseTypesAsParameters,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.UnitTests
+{
+    public class ConsiderPassingBaseTypesAsParametersTests
+    {
+        [Fact]
+        public async Task Unused_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+public class A {}
+public class B : A {}
+
+public class Z
+{
+    public void M(B b)
+    {
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MethodCall_TypeWithBase()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
```

Comment:
Use VerifyCodeFixAsync and pass the same source as fixedSource. The benefits are:

- Whenever a codefix is implemented, the test will fail and require update to add the fixedCode
- Even if there is a case when a codefix shouldn't be offered, it will check that it's indeed not offered, instead of a code action that does nothing is offered.

Reply:
Will do! I have been a little lazy with the implementation of a code-fix but if I manage to save some time this week I will try to work on it.